src/network/networkctl.c

```diff
@@ -2919,6 +2930,488 @@ static int verb_reconfigure(int argc, char *argv[], void *userdata) {
+                        if (r < 0)
+                                return r;
+                        if (r == 0) { /* Child */
+                                execlp("udevadm", "udevadm", "trigger", "--subsystem-match=net", "--action=add",
+                                       "--settle", NULL);
+
+                                log_error_errno(errno, "Failed to execute udevadm: %m");
+                                _exit(EXIT_FAILURE);
+                        }
+                }
+        }
+
+        if (networkd_reload) {
+                _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+
+                if (!networkd_is_running()) {
+                        log_debug("systemd-networkd is not running, skipping reload.");
+                        return 0;
+                }
+
+                r = bus_call_method(bus, bus_network_mgr, "Reload", &error, NULL, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to reload systemd-networkd: %s", bus_error_message(&error, r));
+        }
+
+        return 0;
+}
+
+static int verb_cat(int argc, char *argv[], void *userdata) {
+        _cleanup_(sd_netlink_unrefp) sd_netlink *rtnl = NULL;
+        int r, ret = 0, has_networkd = -1;
+
+        pager_open(arg_pager_flags);
+
+        STRV_FOREACH(name, strv_skip(argv, 1)) {
+                _cleanup_strv_free_ char **dropins = NULL;
+                _cleanup_free_ char *path = NULL;
+                const char *link_config;
+
+                link_config = startswith(*name, "@");
+                if (link_config) {
+                        _cleanup_free_ char *ifname = NULL, *type = NULL;
+
+                        r = parse_link_config(link_config, &ifname, &type);
+                        if (r < 0)
+                                return r;
+
+                        if (streq(type, "network")) {
+                                if (has_networkd < 0)
+                                        has_networkd = networkd_is_running();
+                                if (!has_networkd)
+                                        return log_error_errno(SYNTHETIC_ERRNO(ESRCH),
+                                                               "Cannot cat network file for link if systemd-networkd is not running.");
+
+                                r = get_network_files_by_link(&rtnl, ifname, &path, &dropins);
+                                if (r < 0)
+                                        return ret < 0 ? ret : r;
+                        } else if (streq(type, "link")) {
+                                r = get_link_files_by_link(ifname, &path, &dropins);
+                                if (r < 0)
+                                        return ret < 0 ? ret : r;
+                        } else
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "Invalid config type '%s' for link '%s'.", type, ifname);
```

Comment:
Could you extract the code, say `get_config_files_by_link()`, and reuse it also in verb_edit()?